albatross.ml

```diff
@@ -38,6 +48,35 @@ struct
     in
     Ok (Vmm_trie.fold path t.policies (fun name p acc -> (name, p) :: acc) [])
 
+  let policy_resource_used () t =
+    let root_policy =
+      match policy t with
+      | Ok p -> ( match p with Some p -> p | None -> empty_policy)
+      | Error _ -> empty_policy
+    in
+    let policies = match policies t with Ok p -> p | Error _err -> [] in
+    let vms_used, memory_used, storage_used =
+      List.fold_left
+        (fun (total_vms, total_memory, total_block) (name, policy) ->
+          if name <> Vmm_core.Name.root then
+            ( total_vms + policy.Vmm_core.Policy.vms,
+              total_memory + policy.memory,
+              total_block + match policy.block with Some b -> b | None -> 0 )
+          else (total_vms, total_memory, total_block))
+        (0, 0, 0) policies
+    in
+    Vmm_core.Policy.
+      {
+        vms = root_policy.vms - vms_used;
+        cpuids = root_policy.cpuids;
+        memory = root_policy.memory - memory_used;
+        block =
+          (match root_policy.block with
+          | Some b -> if b > 0 then Some (b - storage_used) else None
+          | None -> None);
```

Comment:
```suggestion
        block = Option.map (fun b -> b - storage_used) root_policy.block;
```

I guess the `> 0` check is sensible, but we don't do it for other values.... I just wish we had an unsigned integer type ;)